backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CBridgeGen.kt

```diff
@@ -834,9 +834,14 @@ private fun KotlinStubs.mapType(
 
     isObjCReferenceType(type) -> ObjCReferenceValuePassing(symbols, type, retained = retained)
 
+    type.isForwardDeclaration() -> TrivialValuePassing(symbols.nativePtrType, CTypes.voidPtr, isForwardDeclaration = true)
```

Comment:
Is it really required to handle this ad hoc? Adjusting the condition covering non-forward structs should be enough. Moreover, that condition should actually handle forward declarations on a general basis, so I guess something is wrong with the IR here.

Reply:
You are right, the problem was in type translation. I pushed the (probably) correct fix and will invite a member of middle-end team to check.

Reply:
Checked ✅